mne/channels/interpolation.py

```diff
@@ -104,8 +104,33 @@ def _make_interpolation_matrix(pos_from, pos_to, alpha=1e-5):
     return interpolation
 
 
+def _do_interp_dots(inst, interpolation, goods_idx, bads_idx):
+    """Dot product of channel mapping matrix to channel data
+    """
+    from mne.io.base import _BaseRaw
+    from mne.epochs import _BaseEpochs
+    from mne.evoked import Evoked
+
+    if isinstance(inst, _BaseRaw):
+        inst._data[bads_idx] = interpolation.dot(inst._data[goods_idx])
+    elif isinstance(inst, _BaseEpochs):
+        tmp = np.dot(interpolation[:, np.newaxis, :],
+                     inst._data[:, goods_idx, :])
+        if np.sum(bads_idx) == 1:
+            tmp = tmp[0]
+        else:
+            tmp = tmp[:, 0, ...]
+        inst._data[:, bads_idx, :] = np.transpose(tmp, (1, 0, 2))
```

Comment:
If you want to get fancy (don't we all?), you can do this same thing in a single line using `np.einsum`. It's a really cool function...